pv_site_production/app.py

```diff
@@ -1,44 +1,104 @@
+            # TODO Make sure we have consistent units for the forecast.
+            # We might want something like "energy" for the given time interval.
+            records.append(
+                {
+                    # TODO Make sure we use the same units everywhere.
+                    "forecast_kw": power / 1000 * 12,
+                    "pv_uuid": pv_id,
+                    # TODO Does it make sense to use the middle of the horizon interval?
+                    "target_datetime_utc": ts
+                    + timedelta(minutes=(horizon[1] + horizon[0]) / 2),
+                }
+            )
+
+    return pd.DataFrame.from_records(records)
 
 
 @click.command()
 @click.option(
-    "--model-name",
-    default="cos",
-    envvar="MODEL_NAME",
-    help="Select which model to use",
-    type=click.STRING,
+    "--config",
+    "-c",
+    "config_path",
+    type=click.Path(path_type=pathlib.Path),
+    help="Config defining the model to use and its parameters.",
+)
+@click.option(
+    "--date",
+    "-d",
+    "timestamp",
+    type=click.DateTime(formats=["%Y-%m-%d-%H-%M"]),
+    default=None,
+    help='Date-time (UTC) at which to make the prediction. Defaults to "now".',
 )
 def run(
-    model_name: str = "cos",
+    config_path: pathlib.Path,
+    timestamp: datetime | None,
 ):
-    """
-    Make app method
+    # logging.basicConfig(level=logging.INFO)
+    """Make app method"""
+    with open(config_path) as f:
+        config = yaml.safe_load(f)
 
-    :param model_name: the model name, current is only 'cos'
-    :return:
-    """
+    if timestamp is None:
+        timestamp = datetime.utcnow()
+
+    get_model = import_from_module(config["run_model_func"])
+
+    _log.debug("Connecting to pv database")
+    url = config["pv_db_url"]
+    pv_db_connection = DatabaseConnection(url=url, base=Base_PV, echo=False)
+
+    # Wrap into a PV data source for the models.
+    _log.debug("Creating PV data source")
+    pv_data_source = DbPvDataSource(pv_db_connection, config["pv_metadata_path"])
+
+    _log.debug("Loading model")
+    model: PvSiteModel = get_model(config, pv_data_source)
+
+    pv_ids = pv_data_source.list_pv_ids()
+    pv_ids = pv_ids[:10]
 
-    # choose model
-    if model_name == "cos":
-        run_model = run_cos_model
-    else:
-        Exception(f"Could not find model {model_name}")
-
-    # run model, this returns a dataframe with the following columns
-    # "target_datetime_utc"
-    # "forecast_kw"
-    # "pv_uuid"
-    results_df = run_model()
-
-    # save model
-    # TODO
-    print(results_df)
+    _log.info("Applying model")
+    # TODO Calling/printing like this to debug but we could call `apply_model` once with all the pv
+    # ids.
+    for pv_id in pv_ids:
+        results_df = apply_model(model, pv_ids=[pv_id], ts=timestamp)
+        print(pv_id)
+        for _, row in results_df.iterrows():
+            print(f"{row['target_datetime_utc']}: {row['forecast_kw']}")
 
```

Comment:
can you leabe the # save model comment in, just so we know where the model is saved. 

I think its nice, if we build up all the results and then pass it though once into a save function

Reply:
But we're not saving the model here. Or you mean saving the results, eventually to the database?